.github/workflows/require-checklist.yml

```diff
@@ -0,0 +1,20 @@
+name: Require Reviewer Checklist
+
+on:
+  pull_request:
+    types: [opened, reopened, edited, synchronize]
+    branches: [master, 1.*]
+
+jobs:
+  require-reviewer-checklist:
+    runs-on: ubuntu-latest
+    steps:
+      - env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          REPO_NAME: ${{ github.event.repository.full_name }}
+          PR_NUMBER: ${{ github.event.number }}
+        run: |
+          if gh pr view "$PR_NUMBER" -R "$REPO_NAME" --json body --jq '.body' | sed -n '/## Reviewer Checklist/,/## After-merge Checklist/p' | grep '\[ \]'; then
```

Comment:
When you don't care about the output from `grep`, it's best to pass `-q`, but that's not a big deal, will approve.